components/script/dom/mutationobserver.rs

```diff
@@ -37,4 +70,166 @@ impl MutationObserver {
+//            options.attributes.set(Some(true));
+        }
+        // Step2: If options’ characterDataOldValue is present and options’ characterData is omitted,
+        // set options’ characterData to true.
+        if options.characterDataOldValue.is_some() && options.characterData.is_some() {
+//            options.characterData.set(Some(true));
+        }
+        // Step3: If none of options’ childList, attributes, and characterData is true, throw a TypeError.
+        if !options.childList && !options.attributes.unwrap_or(false) && !options.characterData.unwrap_or(false) {
+            return Err(Error::Type("childList, attributes, and characterData not true".to_owned()));
+        }
+        // Step4: If options’ attributeOldValue is true and options’ attributes is false, throw a TypeError.
+        if options.attributeOldValue.unwrap_or(false) && !options.attributes.unwrap_or(false) {
+            return Err(Error::Type("attributeOldValue is true but attributes is false".to_owned()));
+        }
+        // Step5: If options’ attributeFilter is present and options’ attributes is false, throw a TypeError.
+        if options.attributeFilter.is_some() && !options.attributes.unwrap_or(false) {
+            return Err(Error::Type("attributeFilter is present but attributes is false".to_owned()));
+        }
+        // Step6: If options’ characterDataOldValue is true and options’ characterData is false, throw a TypeError.
+        if options.characterDataOldValue.unwrap_or(false) && !options.characterData.unwrap_or(false) {
+            return Err(Error::Type("characterDataOldValue is true but characterData is false".to_owned()));
+        }
+        // Step 7
+        let mut registered_not_found: bool = true;
+        for registered in target.registered_mutation_observers().borrow().iter() {
+            if &**registered as *const MutationObserver == self as *const MutationObserver {
+                registered_not_found = false;
+                // TODO: remove matching transient registered observers
+                if let Some(value) = options.attributeOldValue {
+                    registered.attribute_old_value.set(value);
+                }
+                if let Some(value) = options.attributes {
+                    registered.attributes.set(value);
+                }
+                if let Some(value) = options.characterData {
+                    registered.character_data.set(value);
+                }
+                if let Some(value) = options.characterDataOldValue {
+                    registered.character_data_old_value.set(value);
+                }
+                registered.child_list.set(options.childList);
+                registered.subtree.set(options.subtree);
+                *registered.attribute_filter.borrow_mut() = options.attributeFilter.clone().unwrap();
+            }
+        }
+        // Step 8
+        if registered_not_found {
+            target.add_registered_mutation_observer(&self);
```

Comment:
Hi @jdm
I made these changes, but I wasn't able to get it to compile.
It gives the error:
`cannot move out of indexed content`, for the line `observers.borrow()[idx]`

I tried to fix that using `&observers.borrow()[idx]`, but that also gives the following error:
```
error: borrowed value does not live long enough
   --> /home/sumit/Dropbox/NCSU/OODD/FinalProject/servo/components/script/dom/mutationobserver.rs:234:51
    |
234 |             Some(idx) => &(observers.borrow()[idx]),
    |                            ------------------     ^ temporary value dropped here while still borrowed
    |                            |
    |                            temporary value created here
...
251 |     }
    |     - temporary value needs to live until here
```

Also, maybe I am misinterpreting something, but shouldn't this part be in a for loop to iterate over all the observers?